### DISCUSSION TITLE
AWS lambda 권한 요청에 대해

### DISCUSSION THREAD

**Discussion by WaiNaat** [woowacourse-teams/2023-pium#335] [질문있어요] [answered]
# 문제 상황

#331 이슈를 해결하려고 Lambda@Edge를 공부하는 과정에서 저희 IAM 계정에 lambda 권한이 없다는 중대한 사실을 주노에게 들어버렸습니다. 저는 두 개가 다른 줄 알았는데 근본이 같더라구요..

# 이슈 요약

1. 식물 등록이나 식물 이름 검색처럼 작은 이미지로도 충분한 곳에서 큰 이미지를 받아와서 로딩 시간 추가
2. Lambda@Edge를 사용하면 람다함수를 Cloudfront 이벤트에 걸어서 실행 가능
(피움 서비스에서는 Origin Response에 걸면 됨)
![image](https://github.com/woowacourse-teams/2023-pium/assets/77872742/c68588e4-2696-40af-a68a-acdb783f9cae)
3. 해당 방법을 통해 cloudfront에 이미지를 요청할 때 `image.jpg?width=100&height=100&as=webp&quality=90` 이런 느낌으로 조절된 이미지를 반환
4. cloudfront에서는 해당 쿼리 스트링을 기준으로 캐시해서 똑같은 쿼리스트링이 들어왔을 때 캐시된 값을 반환

# 질문

Lambda@Edge를 쓰면 이슈에 적은 내용이 가능하구요
그 외에도 사용자가 본인의 반려식물 사진을 고화질로 올리더라도 그걸 받아올때는 저화질로 받아올 수 있어서 속도 개선이 될 것 같아요

Lambda@Edge를 쓰지 않는다면 S3에 같은 이미지를 원본, 고화질, 저화질처럼 화질별로 분리해서 저장하는 방식으로 이미지 처리가 가능합니다.

lambda 권한을 요청하는 게 좋을까요?

**Comment by bassyu** [comment, marked as answer]
저는 매우 찬성합니다! 
프론트 assets가 아니라서 직접 최적화를 하기엔 어려운 사진들을 원하는 사이즈로 요청하면 매우 좋을 것 같아요~
요청해보고 안되면 어쩔 수 없죠!
